test/necb/unit_tests/tests/test_necb_structures.rb

```diff
@@ -0,0 +1,119 @@
+require_relative '../../../helpers/minitest_helper'
+require_relative '../../../helpers/create_doe_prototype_helper'
+require 'json'
+
+
+# Checks if BTAP::Structure instances are correctly deployed within BTAP.
+class NECB_Structure_Tests < Minitest::Test
+  def test_necb_structures()
+    outd = "output/test_necb_structures"
+    eres = "../expected_results/necb_structures_expected_results.json"
+    tres = "../expected_results/necb_structures_test_results.json"
+    sizd = "sizing_folder"
+
+    @output_folder         = File.join(__dir__, outd)
+    @expected_results_file = File.join(__dir__, eres)
+    @test_results_file     = File.join(__dir__, tres)
+    @sizing_run_dir        = File.join(@output_folder, sizd)
+    @test_results_array    = []
+
+    # Intial test condition.
+    @test_passed = true
+
+    # Range of test options.
+    @templates = [
+      "NECB2011",
+      # "NECB2015",
+      # "NECB2017",
+      # "NECB2020"
+    ]
```

Comment:
Work in progress. But for now, tests results are consistent between NECB vintages, regardless of building type.